tests/JustSaying.UnitTests/Messaging/Channels/SubscriberBusTests/Approvals/WhenListeningWithMultipleGroups.SubscriptionGroups_OverridesDefaultSettingsCorrectly.approved.txt

```diff
@@ -0,0 +1,38 @@
+{
+  "SubscriptionGroups": [
```

Comment:
This file is the output of an [approval test](https://shouldly.readthedocs.io/en/latest/assertions/shouldMatchApproved.html) that is a serialization of the result of interrogating an `ISubscriptionGroupCollection`.

The goal is to ensure that the settings that the group is configured with matches is the same as the _approved_ output, in this file. If something changes such that we break something somewhere between the configuration and the output of the interrogation, then we'll know as the `.received` file generated at test time won't match the approved file.

Reply:
I know that 😉

What I was trying to say is that you can store the "approved output" as an inline C# anonymous object that you serialize on the fly in order to compare, rather than deserializing text from a file.

Reply:
Approval tests where the approved output is stored in a file is a fairly common pattern 🙂 (see https://github.com/approvals/ApprovalTests.Net). If the approved text was stored in an anonymous object, then you'd still have to serialize it as well as the candidate output, and then compare, which opens you up to differences in the way things are serialized, which could cause differences in the output comparison.